stdlib/ast.pyi

```diff
@@ -159,57 +159,94 @@ if sys.version_info >= (3, 8):
     def parse(
         source: str | bytes,
         filename: str | bytes = ...,
-        mode: Literal["exec"] = ...,
         *,
         type_comments: bool = ...,
         feature_version: None | int | _typing.Tuple[int, int] = ...,
     ) -> Module: ...
     @overload
     def parse(
         source: str | bytes,
-        filename: str | bytes = ...,
-        mode: Literal["eval"] = ...,
+        filename: str | bytes,
+        mode: Literal["exec"],
```

Comment:
Not sure why you needed to split this from the previous overload

Reply:
`parse(filename: str = ..., mode: Literal[...])` is impossible as non-default argument cannot follow the default one. If we add `= ...` to `mode` we will return to the situation of the overlapping signatures. If we just remove the first overload (`parse(filename: str = ...)`) than e.g. `r0 = parse()` is not allowed. Did I misunderstand your idea?